src/player/watch_media.rs

```diff
@@ -11,25 +10,44 @@ use crate::{
     media::Media,
     player::mpv::open_mpv,
     player::vlc::open_vlc,
-    TRANSLATION, USE_MPV,
+    TRANSLATION, USE_GECKODRIVER, USE_MPV,
 };
 
 #[tokio::main]
 pub async fn watch_media(media: Media, img_mode: Option<bool>) -> WebDriverResult<()> {
     let language = TRANSLATION.get().unwrap();
     let use_mpv = USE_MPV.get().unwrap();
+    let use_geckodriver = USE_GECKODRIVER.get().unwrap();
 
-    let url = format!("https://vizer.in/{}", &media.url);
-    let mut chromedriver = Command::new("chromedriver").spawn().unwrap();
-    // we need to wait chromedriver to start :(
+    let mut command_name = String::new();
+    if *use_geckodriver {
+        command_name.push_str("geckodriver");
+    } else {
+        command_name.push_str("chromedriver");
+    };
+
+    let mut command = Command::new(command_name)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()
+        .unwrap();
+    // we need to wait command to start :(
     sleep(Duration::from_millis(100));
 
+    let url = format!("https://vizer.in/{}", &media.url);
+
     print!("\x1B[2J\x1B[1;1H");
     println!("{}", language.preparing_misc_text);
 
-    let mut caps = DesiredCapabilities::chrome();
-    caps.set_headless().unwrap();
-    let driver = WebDriver::new("http://localhost:9515", caps).await?;
+    let driver: WebDriver = if *use_geckodriver {
+        let mut caps = DesiredCapabilities::firefox();
+        caps.set_headless().unwrap();
+        WebDriver::new("http://localhost:4444", caps).await?
+    } else {
+        let mut caps = DesiredCapabilities::chrome();
+        caps.set_headless().unwrap();
+        WebDriver::new("http://localhost:9595", caps).await?
```

Comment:
Why you change the port?

Reply:
because the app crashes and give "tcp connection error" if i don't change the port of the DesiredCapabilities::firefox() to "4444"

and also crashes if i don't change the port of the DesiredCapabilities::chrome() to "9595"

Reply:
i tried to change the geckodriver port to accept the 9595 port, by adding a arg to the geckodriver command with "--port 9595", the geckodriver help option says that : 

  ``` -p, --port <PORT>                        Port to use for WebDriver server [default: 4444] ```

but it didn't work